Add analysis-runner to getting_started guide (#44)
populationgenomics/team-docs: getting_started.md

which builds a Batch pipeline from a specific commit in a GitHub repository by
running the driver itself on Hail Batch.

### Analysis runner

The analysis-runner builds a Batch pipeline from a specific commit in a GitHub repository. You can submit a pipeline to run using a CLI tool, which can be installed with:

```bash
conda install -c cpg analysis-runner
```

Make sure you have:

- the name of the [dataset](storage_policies), as this controls what data you can access and which code repositories are available,
- authenticated with your GCP account using `gcloud auth login`,

> - ensured your @populationgenomics.org.au account has been added to the permission group corresponding to the dataset (ask in the #team-software channel if you need to get access).

Reviewer: ```suggestion
- ensured your `@populationgenomics.org.au` account has been added to the permission group corresponding to the dataset (ask in the `#team-software` channel if you need to get access).
```

which builds a Batch pipeline from a specific commit in a GitHub repository by
running the driver itself on Hail Batch.

### Analysis runner

> The analysis-runner builds a Batch pipeline from a specific commit in a GitHub repository. You can interact with an analysis server with the CLI. This can be installed with:

Reviewer: ```suggestion
The analysis-runner builds a Batch pipeline from a specific commit in a GitHub repository. You can submit a pipeline to run using a CLI tool, which can be installed with:
```

```bash
conda install -c cpg analysis-runner
```

Make sure you have:

> - the name of the dataset, as this corresponds to a specific analysis server and approved repository,

Reviewer: ```suggestion
- the name of the dataset, as this controls what data you can access and which code repositories are available,
```

> - authenticated Google Cloud `gcloud auth login`,

Reviewer: ```suggestion
- authenticated with your GCP account using `gcloud auth login`,
```

> - ensured your service account has the correct permissions.

Reviewer: ```suggestion
- ensured your @populationgenomics.org.au account has been added to the permission group corresponding to the dataset (ask in the #team-software channel if you need to get access).
```

Example:

```bash

> # What you would use to run batch directly
> python3 path/to/myscript.py -p 3

Reviewer: ```suggestion
# What you would use to run Batch directly:
# python3 path/to/myscript.py -p 3
```

> # using the analysis runner

Reviewer: ```suggestion
# Using the analysis runner instead:
```

analysis-runner \
    --dataset <dataset> \
    --description "Description of the run" \
    --output-dir gs://<bucket> \

>     --repository hail-batch-test \

Reviewer: ```suggestion
    --repository <repo> \
```

>     --repository hail-batch-test \
>     --commit <hash> \

Reviewer: ```suggestion
```

    path/to/myscript.py -p 3
```

> See the [analysis runner documentation](https://github.com/populationgenomics/analysis-runner/tree/main/cli) for more information

Reviewer: ```suggestion
See the [analysis runner documentation](https://github.com/populationgenomics/analysis-runner/tree/main/cli) for more information.
```

which builds a Batch pipeline from a specific commit in a GitHub repository by
running the driver itself on Hail Batch.

### Analysis runner

> The analysis-runner builds a Batch pipeline from a speific commit in a GitHub repository. You can interact with an analysis server with the CLI. This can be installed with:

Reviewer: ```suggestion
The analysis-runner builds a Batch pipeline from a specific commit in a GitHub repository. You can interact with an analysis server with the CLI. This can be installed with:
```

```bash
conda install -c cpg analysis-runner
```

> Make sure you have the name of the dataset, as this corresponds to a specific analysis server and approved repository.

Reviewer: We should probably expand this a bit and link to the storage policies doc. Before people can access a dataset (and use the runner for it), they need to get added to the corresponding permissions group. Maybe documenting the common failure case (no permissions) here might be helpful.

Author: I've changed this slightly to suggest that permissions are required, but not how you might attain those permissions.
Do you have a suggestion for how to frame this?

You can use the analysis runner CLI in two ways:

> 1. Omitting the `--repository` parameter: use the repository of the local directory that you're in: (a) get the repository name from the git remote; (b) use the commit of HEAD (if the `--commit` parameter is omitted); (c) Make the script path relative to the root of the git repository.

Reviewer: This is pretty detailed and might be a bit overwhelming for new analysts that don't quite understand yet how the whole thing works.

How about just covering the common case here, e.g. "assuming you want to run `python3 myscript.py -p 3`, how do you do this using the analysis runner: commit, then call analysis-runner instead.

For any detailed information (like using another repo) we could just give a link to the documentation in the analysis-server repository. That also avoids duplication and issues with keeping the docs in sync.

Author: Awesome, I'll make those changed, with respect to your other comment now :)
